akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -177,6 +179,9 @@ private[http] class HttpRequestParser(_settings: ParserSettings,
             parseFixedLengthBody(contentLength, closeAfterResponseCompletion)(input, bodyStart)
           }
 
+        case Some(_) if !method.isEntityAccepted ⇒
+          failMessageStart(UnprocessableEntity, s"${method.name} requests must not have an entity")
```

Comment:
Yeah, but since it wouldn't save a single line I went for the duplication instead... :)